### PR TITLE
More fixes to user invitation

### PR DESCRIPTION
**Pull request by allenwq**
- Track who confirmed the invitation
- When the invitation code is used, let user know which email used it
- Further mention user's email when they are added to the course

### CODE REVIEW

File: app/models/course/user_invitation.rb [hunk 1]
```diff
@@ -19,8 +20,9 @@ def self.for_user(user)
     find_by(email: user.emails.confirmed.select(:email))
   end
 
-  def confirm!
+  def confirm!(confirmer:)
```
Comment (weiqingtoh) [inline app/models/course/user_invitation.rb]: The keyword argument here is more for clarity is it?

Comment (allenwq) [inline app/models/course/user_invitation.rb]: yep!

`i.confirm!(confirmer: user)` vs `i.confirm!(user)`

File: config/locales/en/user/registrations.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@ en:
           Already registered? If you have an existing Coursemology account, you can %{sign_in}.
         password_hint: "#{length} characters minimum"
         sign_up: :'user.registrations.new.header'
+        used: 'The invitation code is used by another account, please login that account instead.'
```
Comment (weiqingtoh) [inline config/locales/en/user/registrations.yml]: `The provided invitation code has been claimed using another account, please use that account to access the course instead.`

File: config/locales/en/user/registrations.yml [hunk 3]
```diff
@@ -7,6 +7,8 @@ en:
...
+        used_with_email: 'The invitation code is used by %{email}, please login using that email instead.'
```
Comment (weiqingtoh) [inline config/locales/en/user/registrations.yml]: The provided invitation code has been claimed by %{email}, please use that account to access the course instead.

File: config/locales/en/course/user_registrations.yml [hunk 4]
```diff
@@ -11,5 +11,7 @@ en:
         new_enrol_request_button: 'Request to enrol'
       create:
         invalid_code: 'You have entered an invalid invitation code.'
+        code_taken: 'The invitation code is used by another account, please login that account instead.'
```
Comment (weiqingtoh) [inline config/locales/en/course/user_registrations.yml]: Should we use the same error messages for both?

Comment (weiqingtoh) [inline config/locales/en/course/user_registrations.yml]: `The provided invitation code has been claimed by another account, please use that account to log in instead.`

File: config/locales/en/course/user_registrations.yml [hunk 5]
```diff
@@ -11,5 +11,7 @@ en:
...
+        code_taken_with_email: 'The invitation code is used by %{email}, please login using that email instead.'
```
Comment (weiqingtoh) [inline config/locales/en/course/user_registrations.yml]: `The provided invitation code has been claimed by %{email}, please use that account to log in instead.`

File: config/locales/en/course/mailer/user_added_email.yml [hunk 6]
```diff
@@ -5,3 +5,5 @@ en:
         subject: 'Added to Course %{course}'
         message: >
           You have been added to the course %{course} on %{coursemology}.
+
+          To view the course, please login with %{email}.
```
Comment (weiqingtoh) [inline config/locales/en/course/mailer/user_added_email.yml]: `log in`

**Comment by weiqingtoh** [comment]
After that LGTM.

**Comment by allenwq** [comment]
@weiqingtoh Thanks ! updated !

**Pull request merged**